server/src/main/java/org/candlepin/dto/api/v1/ConsumerDTO.java

```diff
@@ -788,6 +789,7 @@ public ConsumerTypeDTO getType() {
      *
      * @return a reference to this DTO object.
      */
+    @JsonDeserialize(using = ConsumerTypeDeserializer.class)
```

Comment:
What happens when ConsumerDTO.java gets deleted and gets defined in the yaml file? Is there a way to specify custom field se/deserializers for openapi-defined entities?

Reply:
That's indeed a problem that needs to be solved; however is not one that is unique to this addition. We have other such annotations and custom format handlers we'll need to figure out how to add into the (de)serializers.

To some extend this is kind of kicking the can down the road for someone else to sort out, but since we have other areas that use it and need sorting, we can revisit it then.